### PR TITLE
Add coding steps for Lesson 1

### PR DESCRIPTION
**Pull request by maryx**
Adds 5 coding projects for Lesson 1.

### CODE REVIEW

**Comment by maryx** [comment]
@pq - adding you because you reviewed https://github.com/maryx/udacity-course/pull/1 earlier, feel free to punt on it if you're not available.

File: course/02_category_widget/solution_02_category_widget/lib/category.dart [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                          color: Colors.grey[700],
+                          fontSize: 24.0,
+                          fontWeight: FontWeight.w700,
+                        ),
+                  ),
+                ),
+              ),
+            ],
+          ),
+          // Adds inkwell animation when tapped
```
Comment (yjbanov) [inline course/02_category_widget/solution_02_category_widget/lib/category.dart]: nit: did you mean to put this comment one line down?

File: course/02_category_widget/solution_02_category_widget/lib/category.dart [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  const Category({
+    Key key,
+    this.name,
+    this.color,
+    this.iconLocation,
+  })
+      : super(key: key);
+
+  /// Builds a custom widget that shows unit [Category] information.
+  /// This information includes the icon, name, and color for the [Category].
```
Comment (yjbanov) [inline course/02_category_widget/solution_02_category_widget/lib/category.dart]: nit here and elsewhere: the Dart Style guide recommends that the first sentence of a dartdoc is made a standalone paragraph (i.e. separated from the next by 1 blank line).

File: course/03_category_route/solution_03_category_route/lib/category_route.dart [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    Colors.blueAccent,
+    Colors.yellow,
+    Colors.greenAccent,
+    Colors.purpleAccent,
+    Colors.red,
+  ];
+
+  /// Makes the correct number of rows for the Grid View
+  List<Widget> _makeGridRows(List<Category> categories) {
+    // Why do we pass in `_categories.toList()` instead of just `_categories`?
```
Comment (yjbanov) [inline course/03_category_route/solution_03_category_route/lib/category_route.dart]: Is this comment accurate? I do not see `_categories.toList()`. Also, the `categories` parameter is not modified by this method, only read from.

(NOTE: this comment appears several times throughout the PR)

**Comment by yjbanov** [comment]
LGTM w/ comments.

lgtm

File: course/03_category_route/solution_03_category_route/lib/category_route.dart (hunk 3 above)
Comment (maryx) [reply course/03_category_route/solution_03_category_route/lib/category_route.dart]: Thanks for catching! This is an outdated comment. Will resolve through each coding exercise.

File: course/02_category_widget/solution_02_category_widget/lib/category.dart (hunk 1 above)
Comment (maryx) [reply course/02_category_widget/solution_02_category_widget/lib/category.dart]: Thanks!

File: course/02_category_widget/solution_02_category_widget/lib/category.dart (hunk 2 above)
Comment (maryx) [reply course/02_category_widget/solution_02_category_widget/lib/category.dart]: Thanks, Will update the docs.

**Pull request merged**